### DISCUSSION TITLE
mpp-solar project InfluxDB Tabel QPGS0

### DISCUSSION THREAD

**Discussion by AnKloss** [jblance/mpp-solar#123] [Q&A]
Hello all,
For the mpp-solar Project , i just install my Pi 3+ and follow the readme etc..  all good. but simple question , how to get the table QPGS0 including fields ?
May i missing this.  Can i get same info , please .

mfg
Andreas Koss

**Comment by jblance** [comment]
Can have you done? Can you detail the steps so far and results you have got.
Also useful to know what inverter and setup you have.

**Reply by AnKloss** [reply]
Hello
i follow the 
1) MQTT_Influx_Grafana.md including testing,
2) last step i do >> sudo pip install mpp-solar
3) install mpp-solar.servus
so far so good, but in telegram i get error , as cannot find the db felds

**Reply by AnKloss** [reply]
Inverters PIP Max 8048,  2 parallel

**Comment by jblance** [comment]
Did mppsolar work on the commandline?
What is the status of the various services?

**Reply by AnKloss** [reply]
Last login: Thu Jun 17 06:08:31 2021 from 192.168.222.100
ubuntu@ubuntu:~$ systemctl --user status mpp-solar
● mpp-solar.service - MPP Solar Service
     Loaded: loaded (/etc/xdg/systemd/user/mpp-solar.service; enabled; vendor preset: enabled)
     Active: active (running) since Thu 2021-06-17 05:58:50 UTC; 1h 9min ago
   Main PID: 1761 (python3)
     CGroup: /user.slice/user-1000.slice/user@1000.service/mpp-solar.service
             └─1761 /usr/bin/python3 /usr/local/bin/mpp-solar -C /etc/mpp-solar/mpp-solar.conf --daemon

Jun 17 07:07:50 ubuntu python3[1761]: Sleeping for 5 sec
Jun 17 07:07:55 ubuntu python3[1761]: Getting results from device: mppsolar device - name: Inverter_1, port: <mppsolar.>
Jun 17 07:07:57 ubuntu python3[1761]: Getting results from device: mppsolar device - name: Inverter_2, port: <mppsolar.>
Jun 17 07:07:59 ubuntu python3[1761]: Sleeping for 5 sec
Jun 17 07:08:04 ubuntu python3[1761]: Getting results from device: mppsolar device - name: Inverter_1, port: <mppsolar.>
Jun 17 07:08:06 ubuntu python3[1761]: Getting results from device: mppsolar device - name: Inverter_2, port: <mppsolar.>
Jun 17 07:08:08 ubuntu python3[1761]: Sleeping for 5 sec
Jun 17 07:08:13 ubuntu python3[1761]: Getting results from device: mppsolar device - name: Inverter_1, port: <mppsolar.>
Jun 17 07:08:16 ubuntu python3[1761]: Getting results from device: mppsolar device - name: Inverter_2, port: <mppsolar.>
Jun 17 07:08:18 ubuntu python3[1761]: Sleeping for 5 sec

**Reply by jblance** [reply]
What are the results just using mppsolar on the command line?

**Reply by AnKloss** [reply]
mppsolar work on the commandline, yes


ubuntu@ubuntu:~$  mpp-solar -p /dev/hidraw0 -c QPI -I
2021-06-17 07:13:58,571:INFO:__init__:main@235: Solar Device Command Utility, version: 0.7.67, move jk04 to new processing, add lookup functionality
2021-06-17 07:13:58,572:INFO:__init__:main@338: Creating device "unnamed" (type: "mppsolar") on port "/dev/hidraw0 (porttype=None)" using protocol "PI30"
2021-06-17 07:13:58,577:INFO:device:set_port@142: Using hidrawio for communications
2021-06-17 07:13:58,602:INFO:__init__:main@394: Looping 1 commands
2021-06-17 07:13:58,603:INFO:__init__:main@404: Getting results from device: mppsolar device - name: unnamed, port: <mppsolar.io.hidrawio.HIDRawIO object at 0xb687afe8>, protocol: <mppsolar.protocols.pi30.pi30 object at 0xb687ae68> for command: QPI, tag: QPI, outputs: screen
2021-06-17 07:13:58,603:INFO:device:run_command@270: Running command QPI
2021-06-17 07:13:58,603:INFO:abstractprotocol:get_full_command@30: Using protocol b'PI30' with 36 commands
2021-06-17 07:13:58,604:INFO:device:run_command@296: full command b'QPI\xbe\xac\r' for command QPI
2021-06-17 07:13:58,604:INFO:__init__:get_outputs@26: attempting to create output processor: screen
2021-06-17 07:13:58,611:INFO:screen:output@19: Using output processor: screen
Parameter                       Value           Unit
error                           USB open error: [Errno 2] No such file or directory: '/dev/hidraw0'

**Reply by AnKloss** [reply]
i see i get error in USB, will fix it now

**Comment by jblance** [comment]
So that is failing, no point messing with the service until you get it
working

What connectivity are you using?

**Reply by AnKloss** [reply]
USB to RS232 Serial Port DB9 9 Pin Male COM Port Converter Adapter Cable

**Reply by AnKloss** [reply]
question, if get connect , the the DB will by show up in InfluxDB ?

**Comment by jblance** [comment]
Will need to create a db
Telegraf and influx are just being used in standard ways, but you need to
get info sending first and it is easier to find the problems by doing it
manually first

If you are using a USB to 232 converter they normally show up as a ttyUSBx
device

**Reply by AnKloss** [reply]
okay thanks,  will see why this usb do this.

can you send my the structure of the DB , i copy it and do it later the is get the right data

mfg
Andreas Kloss

**Comment by jblance** [comment]
Influx is a timeseries db, so no real structure. The fields and tags depend
on what messages you send via telegraf

Best to get messages sending first.

I personally don't use influx and grafana anymore, so can't show you my db
structure

**Reply by AnKloss** [reply]
and what you use now ?

**Reply by AnKloss** [reply]
if a DB not use structure, i think is not safety at all

**Comment by jblance** [comment]
It's not the way time series dbs work. They're not like a sql db

I use home assistant for info display and automation
